packages/ckeditor5-html-support/src/integrations/image.js

```diff
@@ -110,19 +113,40 @@ function viewToModelImageAttributeConverter( dataFilter ) {
 				}
 			}
 
-			// For a block image, we want to preserve the attributes on our own.
-			// The inline image attributes will be handled by the GHS automatically.
 			function preserveLinkAttributes( viewContainerElement ) {
 				if ( data.modelRange && data.modelRange.getContainedElement().is( 'element', 'imageBlock' ) ) {
 					preserveElementAttributes( viewContainerElement, 'htmlLinkAttributes' );
 				}
+			}
+		}, { priority: 'low' } );
+	};
+}
+
+// View-to-model conversion helper preserving allowed attributes on {@link module:table/table~Table Table}
+// feature model element from figure view element.
+//
+// @private
+// @param {module:html-support/datafilter~DataFilter} dataFilter
+// @returns {Function} Returns a conversion callback.
+function viewToModelFigureAttributeConverter( dataFilter ) {
+	return dispatcher => {
+		dispatcher.on( 'element:figure', ( evt, data, conversionApi ) => {
+			const viewFigureElement = data.viewItem;
+
+			if ( !data.modelRange || !viewFigureElement.hasClass( 'image' ) ) {
+				return;
+			}
+
+			preserveElementAttributes( viewFigureElement, 'htmlFigureAttributes' );
```

Comment:
I guess that we don't need this to be in a separate function.

Reply:
My reasoning was to keep it the same as in image/table/media converter, so if anyone decide it would be better to extract this peace of code to util - it would be easier to spot and extract, but no problem - I will inline this.